testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/Standard.java

```diff
@@ -184,6 +185,61 @@ else if (logicalName.startsWith(WORKER_NTH)) {
         enableTrinoJavaDebugger(dockerContainer, logicalName, debugPort);
     }
 
+    private static void enableTrinoJmxRmi(DockerContainer dockerContainer)
+    {
+        String logicalName = dockerContainer.getLogicalName();
+
+        int jmxPort;
+        if (logicalName.equals(COORDINATOR)) {
+            jmxPort = 6005;
+        }
+        else if (logicalName.equals(WORKER)) {
+            jmxPort = 6009;
+        }
+        else if (logicalName.startsWith(WORKER_NTH)) {
+            int workerNumber = parseInt(logicalName.substring(WORKER_NTH.length()));
+            jmxPort = 6008 + workerNumber;
+        }
+        else {
+            throw new IllegalStateException("Cannot enable Java debugger for: " + logicalName);
+        }
+
+        enableTrinoJmxRmi(dockerContainer, logicalName, jmxPort);
+    }
+
+    private static void enableTrinoJmxRmi(DockerContainer container, String containerName, int jmxPort)
+    {
+        log.info("Enabling Java JMX RMI for container: '%s' on port %d", containerName, jmxPort);
+
+        try {
+            FileAttribute<Set<PosixFilePermission>> rwx = PosixFilePermissions.asFileAttribute(PosixFilePermissions.fromString("rwxrwxrwx"));
+            Path script = Files.createTempFile("enable-java-jmx-rmi", ".sh", rwx);
+            script.toFile().deleteOnExit();
+            Files.writeString(
+                    script,
+                    format(
+                            "#!/bin/bash\n" +
+                                    "printf '%%s\\n' '%s' >> '%s'\n",
+                                    """
+                                    -Dcom.sun.management.jmxremote=true
```

Comment:
the use of format, singe line string and multiline string makes me wonder what the file contents gonna be

also, use of format + manual formatting (replacement of __PORT__)... maybe just using one formatting layer would be enough?